lib/app/modules/quality_reporting/submit_data_sequence.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require_relative '../../utils/measure_operations'
+require_relative '../../utils/bundle'
+
+module Inferno
+  module Sequence
+    class SubmitDataSequence < SequenceBase
+      include MeasureOperations
+      include BundleParserUtil
+      title 'Submit Data'
+
+      test_id_prefix 'submit_data'
+
+      description 'Ensure that resources relevant to a measure can be submitted via the $submit-data operation'
+
+      test 'Submit Data valid submission' do
+        metadata do
+          id '01'
+          link 'https://www.hl7.org/fhir/measure-operation-submit-data.html'
+          desc 'Submit resources relevant to a measure, and then verify they persist on the server.'
+        end
+
+        # Get the patient data to submit. For now this just always uses CMS130 data.
+        patient_file = File.expand_path('../../../../resources/quality_reporting/CMS130/Bundle/cms130-patient-bundle.json', __dir__)
+        patient_bundle = FHIR::STU3::Json.from_json(File.read(patient_file))
+        resources = patient_bundle.entry.map(&:resource)
+        patient = resources.first { |r| r.resourceType == 'Patient' }
+
+        # Submit the data
+        submit_data_response = submit_data(@instance.measure_to_test, patient.id, '2019', '2019', resources)
+        assert_response_ok(submit_data_response)
+
+        # GET and assert presence of all submitted resources
+        resources.each do |r|
+          relative_url = "#{r.resourceType}/#{r.id}"
+          get_response = @client.get(relative_url)
+          assert_response_ok(get_response, "Submitted resource unavailable on the server: #{relative_url}")
+        end
```

Comment:
This block of code will assert that all the resources present in the Patient bundle exist on the server identified by the IDs in the Patient Bundle. Two things:

1) This will fail if the server generates the ID for us. I think what we should be doing is examining `submit_data_response`'s payload and getting the `location`s for each resource and sending gets to those instead. Which leads me into 2
2) This block of code doesn't actually assert that the `MeasureReport` was successfully posted, as the `MeasureReport` is something that you create yourself rather than reading in from the bundle, so it's not part of this loop. However if you parse the payload from the response and use the IDs from that, we will get the `MeasureReport` GET request for free.

Reply:
1. Good point about the IDs. I don't think we should use the `location` of the resources in the response, because although CQF Ruler returns this in the `$submit-data` response, the spec doesn't actually define any response body as far as I can tell (have to constantly remind myself we want to test conformance to the spec, not get CQF Ruler to pass). Maybe this is where we use resource `identifier`s to verify their presence?
2. The same concern about the response applies. The `MeasureReport` doesn't have an `identifier` but we could give it one. What do you think?

Reply:
True, returning the locations is a HAPI thing so it's not guaranteed for all systems under test. Are we ok with the very very rare case of something maybe not having an identifier?

Reply:
I think so. I've talked about that with Sam before and he seemed to think that was an acceptable assumption.

Reply:
But also we control the test data. So if we want everything to have an `identifier`, we can make it so.

Reply:
yeah let's go with Identifier searching then